internal/deploy/bootstrap.go

```diff
@@ -63,16 +65,97 @@ func Bootstrap(
 		manifests = append(manifests, []byte(wildCardRoleAndAssignment)...)
 	}
 
-	if err := applyManifests(
+	manifestObjs, err := parseManifests(k8s, manifests, dryRun)
+	if err != nil {
+		return false, err
+	}
+	if err = PatchDeploymentWithInKcpModeFlag(manifestObjs, isInKcpMode); err != nil {
+		return false, err
+	}
+	err = applyManifests(
 		ctx, k8s, manifests, applyOpts{
-			dryRun, force, defaultRetries, defaultInitialBackoff},
-	); err != nil {
+			dryRun, force, defaultRetries, defaultInitialBackoff}, manifestObjs,
+	)
+	if err != nil {
 		return false, err
 	}
 
 	return hasKyma(string(manifests))
 }
 
+func PatchDeploymentWithInKcpModeFlag(manifestObjs []ctrlClient.Object, isInKcpMode bool) error {
+	var deployment *appsv1.Deployment
+	for _, manifest := range manifestObjs {
+		manifestJSON, err := getManifestJSONForDeployment(manifest)
+		if err != nil || manifestJSON == nil {
+			return err
+		}
+		deployment, err = getDeployment(manifestJSON)
+		if err != nil {
+			return err
+		}
+		deploymentArgs := deployment.Spec.Template.Spec.Containers[0].Args
+		hasKcpFlag := checkDeploymentHasKcpFlag(deploymentArgs)
+		if !hasKcpFlag && isInKcpMode {
```

Comment:
If this function is called with `isInKcpMode` false, nothing will happen, so why have this parameter even? The caller should decide to call this if needed.

Reply:
yes makes sense, will only call the function if isInKcpMode is true